### PR TITLE
Routers should be able to override the host value on Routes

### PR DESCRIPTION
**Pull request by smarterclayton**
Add `--hostname-template` which takes a format like `${name}-${namespace}.myapps.com` for defaulting and add `--override-hostname` to allow forcing of the route.

[test]

### CODE REVIEW

File: pkg/cmd/infra/router/router.go [hunk 1]
```diff
@@ -47,9 +55,38 @@ func (o *RouterSelection) Bind(flag *pflag.FlagSet) {
...
+		}
+		return s
+	}
+}
+
 // Complete converts string representations of field and label selectors to their parsed equivalent, or
 // returns an error.
 func (o *RouterSelection) Complete() error {
+	if len(o.HostnameTemplate) == 0 && o.OverrideHostname {
+		return fmt.Errorf("--override-hostname requires that --hostname-template be specified")
```
Comment (mfojtik) [inline pkg/cmd/infra/router/router.go]: is specified?

Comment (Miciah) [inline pkg/cmd/infra/router/router.go]: "be" is correct (subjunctive mood).

**Comment by smarterclayton** [comment]
@ramr / @pravisankar can one of you look through this?  Still need to add the integration test, but is functional.

**Comment by ramr** [comment]
Yeah, some tests would be good - the changes look good.

**Comment by smarterclayton** [comment]
[test][extended:core(HAProxy)]

**Comment by smarterclayton** [comment]
[test][extended:core]

**Comment by smarterclayton** [comment]
Ram, can you help me debug the extended failure?  Getting pulled into something else today - would like to close this out.

**Comment by ramr** [comment]
@smarterclayton  sure - will take a look at it this eve/over the weekend.

**Comment by ramr** [comment]
@smarterclayton  took a while - this commit https://github.com/ramr/origin/commit/9787405eda835faf2de9c3272fb7d34cee737e3f fixes the test errors but it also needs a rebase from master as there was another fix that went in last week to properly report 503 pages (status headers).

**Comment by smarterclayton** [comment]
Thanks, much appreciated

**Comment by ramr** [comment]
@smarterclayton if you are swamped, did you want me to create a new PR from my branch with your code as the base and rebased from master + the above commit (https://github.com/ramr/origin/commit/9787405eda835faf2de9c3272fb7d34cee737e3f) added in?  Let me know. Thx

**Comment by ramr** [comment]
nm - should have reloaded the page before I commented!! :^( ... think me needs that coffee now!!

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
Trap is green [merge]

**Pull request merged**